### DISCUSSION TITLE
Github CI/CD Pipeline with Unit Testing

### DISCUSSION THREAD

**Discussion by nevermore23274** [TheSirLancelot/Netulla#28] [Ideas]
Something worth some consideration: 
https://blog.streamlit.io/host-your-streamlit-app-for-free/
Or: 
https://dev.to/harshi606/deploying-streamlit-apps-for-free-11f1

I saw on the ticket there's a link ( https://discuss.streamlit.io/t/hosting-streamlit-on-github-pages/356 ) however that was from 2019. I do think this would save us quite a bit of effort, would also be a check that you didn't accidently break another feature when you were coding.

**Comment by TheSirLancelot** [comment]
Lol this solves both of the idea tickets I put in the icebox. I love it.

Essentially whenever a push or merge would happen, we would run our unit tests using a GitHub workflow (e.g. we spin up a docker container, connect to it, run our unit tests). If they all pass, the hosting site sees the update and then hosts the new updates for us.
